src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs

```diff
@@ -165,7 +165,13 @@ private HttpRequestMessage CreateAndGetHttpRequestMessage(string functionName, s
 
         private void AddRequestHeadersAndSetRequestUri(HttpRequestMessage httpRequestMessage, string functionName, string invocationId)
         {
-            httpRequestMessage.RequestUri = new Uri(new UriBuilder(WorkerConstants.HttpScheme, WorkerConstants.HostName, _httpWorkerOptions.Port, functionName).ToString());
+            string pathValue = functionName;
+            // _httpWorkerOptions.Type is populated only in customHandler section
+            if (httpRequestMessage.RequestUri != null && !string.IsNullOrEmpty(_httpWorkerOptions.Type))
+            {
+                pathValue = httpRequestMessage.RequestUri.AbsolutePath;
```

Comment:
`System.Uri.AbsolutePath` doesn't contain the query string. If forwarding the request to the handler, I think we want to include the query string as well.

```suggestion
                pathValue = httpRequestMessage.RequestUri.PathAndQuery;
```

Reply:
Query string is appended here: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs#L160